File: isegoria/src/main/java/com/eulersbridge/isegoria/network/api/models/Institution.java
package com.eulersbridge.isegoria.network.api.models;

import android.text.TextUtils;

import com.squareup.moshi.Json;

@SuppressWarnings("unused")
public class Institution {

	@Json(name = "institutionId")
	public long id;

	public long newsFeedId;

	// Different JSON names used depending on API endpoint.
    // Try to read both, keep private and use public getter to return whichever is non-null.
	private String institutionName;
	private String name;

	public String state;
	public String campus;
	public String country;

	public String getName() {
		if (TextUtils.isEmpty(institutionName)) {
			return name;

		} else {
			return institutionName;
		}
	}

	@Override
	public String toString() {
		return getName();
	}
}

File: isegoria/src/main/kotlin/com/eulersbridge/isegoria/network/api/model/Institution.kt
package com.eulersbridge.isegoria.network.api.model

import com.squareup.moshi.Json

data class Institution (
    @field:Json(name = "institutionId") var id: Long = 0,
    var newsFeedId: Long = 0,

    // Different JSON names used depending on API endpoint.
    // Try to read both, keep private and use public getter to return whichever is non-null.
    private val institutionName: String?,
    private val name: String?,

    var state: String?,
    var campus: String?,
    var country: String?
) {
    fun getName(): String? {
        return if (institutionName.isNullOrBlank()) {
            name

        } else {
            institutionName
        }
    }

    override fun toString(): String {
        return getName() ?: ""
    }
}
